src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -71,12 +75,43 @@ public ActiveSolutionBoundTracker(IHost host, IActiveSolutionTracker activeSolut
             this.ProjectKey = this.solutionBindingInformationProvider.GetProjectKey();
         }
 
+
         private void OnActiveSolutionChanged(object sender, EventArgs e)
         {
+            UpdateConnection();
```

Comment:
A solution has been opened or closed, so we need to connect/disconnect as appropriate.